delphi/latents/constructors.py

```diff
@@ -109,6 +109,40 @@ def pool_max_activation_windows(
     return token_windows, activation_windows
 
 
+def is_single_token_feature(
+    activations: Float[Tensor, "examples ctx_len"],
+    quantile_threshold: float = 0.5,
+    activation_ratio_threshold: float = 0.8,
+) -> bool:
+    """
+    Determine if a feature is primarily activated by single tokens.
+
+    Args:
+        activations: Activation values across context windows
+        quantile_threshold: Threshold for considering top activations (0.5 means top 50%)
+        activation_ratio_threshold: Ratio of single-token activations needed (0.8 means 80%)
+
+    Returns:
+        bool: True if the feature is primarily single-token activated
+    """
+    # For each example, check if activation is concentrated in a single position
+    max_activations = activations.max(dim=1).values
+    top_k = int(len(max_activations) * quantile_threshold)
+    top_indices = max_activations.topk(top_k).indices
```

Comment:
I believe that the activations are already in top k order (tbh they should probably be renamed to ordered_example_acts or similar to reflect this). So we should be able to directly slice the first `len(activations) * quantile_threshold` activations.